subsys/emds/emds_flash_new.c

```diff
@@ -49,3 +49,16 @@ int emds_flash_init(struct emds_partition *partition)
 
 	return 0;
 }
+
+int emds_flash_erase_partition(const struct emds_partition *partition)
+{
+	const struct flash_area *fa = partition->fa;
+	int rc;
+
+	rc = flash_area_erase(fa, 0, fa->fa_size);
```

Comment:
erase or flatten? On e.g. nrf54l15 if you call this the function will instantly return and do nothing. If you just care about being able to write to a sector then it's not an issue but if you are calling this to e.g. prevent old data being read as current in future or after a reboot then this is broken